geth/account/accounts_test.go

```diff
@@ -103,9 +107,381 @@ func TestVerifyAccountPasswordWithAccountBeforeEIP55(t *testing.T) {
 	err = common.ImportTestAccount(keyStoreDir, "test-account3-before-eip55.pk")
 	require.NoError(t, err)
 
-	acctManager := account.NewManager(nil)
+	accManager := NewManager(nil)
 
 	address := gethcommon.HexToAddress(TestConfig.Account3.Address)
-	_, err = acctManager.VerifyAccountPassword(keyStoreDir, address.Hex(), TestConfig.Account3.Password)
+	_, err = accManager.VerifyAccountPassword(keyStoreDir, address.Hex(), TestConfig.Account3.Password)
 	require.NoError(t, err)
 }
+
+var (
+	testErrWhisper  = errors.New("Can't return a whisper service")
+	testErrKeyStore = errors.New("Can't return a key store")
+)
+
+func TestManagerTestSuite(t *testing.T) {
+	nodeManager := newMockNodeManager(t)
+	accManager := NewManager(nodeManager)
+
+	keyStoreDir, err := ioutil.TempDir(os.TempDir(), "accounts")
+	require.NoError(t, err)
+	keyStore := keystore.NewKeyStore(keyStoreDir, keystore.LightScryptN, keystore.LightScryptP)
+	defer os.RemoveAll(keyStoreDir) //nolint: errcheck
+
+	testPassword := "test-password"
+
+	// Initial test - create test account
+	nodeManager.EXPECT().AccountKeyStore().Return(keyStore, nil)
+	addr, pubKey, mnemonic, err := accManager.CreateAccount(testPassword)
+	require.NoError(t, err)
+	require.NotNil(t, addr)
+	require.NotNil(t, pubKey)
+	require.NotNil(t, mnemonic)
+
+	s := &ManagerTestSuite{
+		testAccount: testAccount{
+			"test-password",
+			addr,
+			pubKey,
+			mnemonic,
+		},
+		nodeManager:    nodeManager,
+		accManager:     accManager,
+		keyStore:       keyStore,
+		shh:            whisper.New(nil),
+		gethAccManager: accounts.NewManager(),
+	}
+
+	suite.Run(t, s)
+}
+
+func newMockNodeManager(t *testing.T) *common.MockNodeManager {
+	ctrl := gomock.NewController(t)
+	return common.NewMockNodeManager(ctrl)
+}
+
+type ManagerTestSuite struct {
+	suite.Suite
+	testAccount
+	nodeManager    *common.MockNodeManager
+	accManager     *Manager
+	keyStore       *keystore.KeyStore
+	shh            *whisper.Whisper
+	gethAccManager *accounts.Manager
+}
+
+type testAccount struct {
+	password string
+	address  string
+	pubKey   string
+	mnemonic string
+}
+
+// reinitMock is for reassigning a new mock node manager to account manager.
+// Stating the amount of times for mock calls kills the flexibility for
+// development so this is a good workaround to use with EXPECT().AnyTimes()
+func (s *ManagerTestSuite) reinitMock() {
+	s.nodeManager = newMockNodeManager(s.T())
+	s.accManager.nodeManager = s.nodeManager
+}
+
+func (s *ManagerTestSuite) TestCreateAndRecoverAccount() {
+	s.reinitMock()
+
+	// Don't fail on empty password
```

Comment:
How is this connected to `TestCreateAndRecoverAccount`? I am not sure this belongs here...

Reply:
Since the account is created in `TestManagerTestSuite`, this one is just testing two more cases for account creation. And it also tests recovering of the previously created account plus a failing case.

Reply:
The thing is that this test case has nothing to do with recovering an account. Maybe let's split it into `TestCreateAccount` and `TestRecoverAccount`? 

In `TestCreateAccount` we can have an empty password case and failed `CreateAccount()` call:
```go
s.nodeManager.EXPECT().AccountKeyStore().Return(nil, testErrKeyStore)
_, _, _, err = s.accManager.CreateAccount(s.password)
s.Equal(err, testErrKeyStore)
```

In `TestRecoverAccount`, we will just test recovering an account, so only `accManager.RecoverAccount()` calls. Does it sound ok?

Reply:
Surely, splitting.